src/org/traccar/api/MediaFilter.java

```diff
@@ -54,34 +54,32 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 }
             }
             if (userId == null) {
-                throw new NotAuthorizedException("Not authorized");
+                ((HttpServletResponse) response).setStatus(HttpServletResponse.SC_UNAUTHORIZED);
```

Comment:
You can cast it once at the beginning of the method. It will probably make code easier to read.